urpautils/file_utils.py

```diff
@@ -0,0 +1,235 @@
+"""Module containing universal functions for file operations that can be used with urpa robots"""
+
+import glob
+import logging
+import os
+import shutil
+import time
+
+from typing import Optional
+from urpautils.universal import timestamp
+
+import __main__
+
+logger = logging.getLogger(__name__)
+
+
+def remove_dir(path: str) -> None:
+    """Removes a directory
+
+    :param path:    path
+    :return:        None
+    """
+    if os.path.isdir(path):
+        logger.info(f"Removing directory '{path}'")
+        shutil.rmtree(path)
+
+
+def remove(path: str) -> None:
+    """Removes a file
+
+    :param path:    path
+    :return:        None
+    """
+    if os.path.isfile(path):
+        os.remove(path)
+
+
+def move(file_path: str, destination: str) -> None:
+    """Moves a file
+
+    :param file_path:    path
+    :param destination:  destination path
+    :return:             None
+    """
+    if os.path.isfile(file_path):
+        shutil.move(file_path, destination)
+
+
+def copy(src: str, dest: str) -> None:
+    """Copies a file
+
+    :param src:   source path
+    :param dest:  destination path
+    :return: None
+    """
+    shutil.copyfile(src, dest)
+
+
+def remove_files_older_than(dir_path: str, days: int) -> None:
+    """Removes all files in a directory that are older than 'days' days
+
+    :param dir_path:     path
+    :param days:         number of days
+    :return:             None
+    """
+    files = os.listdir(dir_path)
+    for file in files:
+        file_path = os.path.join(dir_path, file)
+        if os.path.isfile(file_path):
+            # file age in seconds
+            file_age = time.time() - os.stat(file_path).st_mtime
+            # file age in days
+            file_age /= 86400
+            if file_age > days:
+                logger.info(f"Removing '{file_path}' because it is older than '{days}' days")
+                remove(file_path)
+
+
+def write_txt_file(file_name: str, content: str, mode: str = "w", encoding: str = "utf-8-sig") -> None:
+    """Writes a text file
+
+    :param file_name:    path
+    :param content:      string to write
+    :param mode:         mode of writing (writing, appending, ...)
+    :param encoding:     encoding to use
+    :return:             None
+    """
+    possible_modes = ("w", "a", "w+", "a+")
+    if not mode in possible_modes:
+        raise ValueError(f"Invalid write mode '{mode}'. Please use one of the following: '{possible_modes}'")
+    with open(file_name, mode, encoding=encoding) as txt_file:
+        txt_file.write(content)
+
+
+def read_txt_file(file_name: str, encoding: str = "utf-8-sig") -> Optional[str]:
+    """Reads a text file
+
+    :param file_name:    path
+    :param encoding:     encoding to use
+    :return:             str content
+    """
+    if not os.path.isfile(file_name):
+        return None
```

Comment:
This construct could lead to problem when you need to distinguished between empty file and non-existing file.